doc/source/conf.py

```diff
@@ -63,39 +68,174 @@
 # extensions coming with Sphinx (named 'sphinx.ext.*') or your custom
 # ones.
 extensions = [
-    "nbsphinx",
     "sphinx.ext.autodoc",
-    "sphinx.ext.autosummary",
-    "sphinx.ext.doctest",
-    # 'sphinx.ext.intersphinx',
-    # 'sphinx.ext.linkcode',
-    "numpydoc",
+    "autoapi.extension",
+    "nbsphinx",
+    "sphinx.ext.intersphinx",
+    "sphinx.ext.linkcode",
+    "sphinx.ext.napoleon",
     "matplotlib.sphinxext.plot_directive",
     "myst_parser",
 ]
+autodoc_typehints = "both"  # signature and description
+
+napoleon_numpy_docstring = True
+napoleon_include_init_with_doc = False
+napoleon_include_private_with_doc = False
+napoleon_include_special_with_doc = True
+napoleon_use_admonition_for_examples = False
+napoleon_use_admonition_for_notes = False
+napoleon_use_admonition_for_references = False
+napoleon_use_ivar = False
+napoleon_use_param = True
+napoleon_use_rtype = True
+napoleon_use_keyword = True
+napoleon_custom_sections = None
+
+_base_path = os.path.abspath(os.path.join(__file__, "../../.."))
+_path_cache: Dict[str, bool] = {}
+
+
+def _check_path(path: str) -> bool:
+    """Check if a path exists and cache the result."""
+    path = os.path.join(_base_path, path)
+    if path in _path_cache:
+        result = _path_cache[path]
+    else:
+        result = os.path.exists(path)
+        _path_cache[path] = result
+    return result
+
+
+def linkcode_resolve(domain: str, info: Dict[str, str]):
+    """linkcode extension override to link to the source code on GitHub."""
+    if domain != "py":
+        return None
+    if not info["module"]:
+        return None
+    if not info["module"].startswith("mlos_"):
+        return None
+    package = info["module"].split(".")[0]
+    filename = info["module"].replace(".", "/")
+    path = f"{package}/{filename}.py"
+    if not _check_path(path):
+        path = f"{package}/{filename}/__init__.py"
+        if not _check_path(path):
+            warning(f"linkcode_resolve failed to find {path}")
+            warning(f"linkcode_resolve info: {json.dumps(info, indent=2)}")
+    return f"https://github.com/microsoft/MLOS/tree/main/{path}"
+
+
+def is_on_github_actions():
+    """Check if the documentation is being built on GitHub Actions."""
+    if "CI" not in os.environ or not os.environ["CI"] or "GITHUB_RUN_ID" not in os.environ:
+        return False
+    return True
```

Comment:
I think we can write just
```suggestion
    return os.environ.get("CI") and os.environ.get("GITHUB_RUN_ID")
```